### PR TITLE
fix: unified Share Action types

### PR DESCRIPTION
**Pull request by juniorklawa**
## Summary:
This pull request simplifies the type and fixes a TypeScript object destructuring problem by combining the ShareSharedAction and ShareDismissedAction types into a single ShareAction type. Enabling proper access to the activityType property without TypeScript errors.

## Changelog:
[INTERNAL] [FIXED] -  Unified ShareSharedAction and ShareDismissedAction into ShareAction. Resolved TypeScript issue with object destructuring.

## Test Plan:

With the current typing:
image

With the proposed solution:
image

### CODE REVIEW

**Comment by philIip** [comment]
love it

File: packages/react-native/Libraries/Share/Share.d.ts [hunk 1]
```diff
@@ -27,17 +27,11 @@ export type ShareOptions = {
   anchor?: number | undefined;
 };
 
-export type ShareSharedAction = {
-  action: 'sharedAction';
+export type ShareAction = {
```
Comment (NickGerleman) [inline packages/react-native/Libraries/Share/Share.d.ts:30]: After this change, it is valid to check for `activityType`, but it is always null. So, we should either:
1. Keep this as-is, to require discriminating against the `action` field before checking `activityType`.
1. Add `null` to the list of possible types of `activityType` (unlike Flow, TS optionals may not be null).

Comment (juniorklawa) [reply packages/react-native/Libraries/Share/Share.d.ts:30]: Hey Nick! Thanks for the review. You're right, I'm going to add null to the activityType typing, It aligns better with other types used in the codebase (this "... | null | undefined" approach), and also avoids making the type a little more complex.

**Review by NickGerleman — APPROVED** [review]

**Comment by philIip** [comment]
@juniorklawa can you rebase and retrigger the CI jobs?

**Comment by juniorklawa** [comment]
@philIip sorry, i got lost here, should i do something else?

**Comment by philIip** [comment]
nope should all be done! thanks for your contribution!

**Pull request closed without merging**